### PR TITLE
WFLY-11404 Add journal-file-open-timeout attribute to messaging server

### PR DESCRIPTION
**Pull request by TomasHofman**
https://issues.jboss.org/browse/WFLY-11404

Adds journal-file-open-timeout attribute to activemq messaging server. The parameter has already been defined on activemq side, this will allow it to be configured from Wildfly configuration.

Please make sure your PR meets the following requirements:
- [x] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue(s)
- [x] Pull Request contains description of the issue(s)
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted

For bigger changes, major and minor component upgrades make sure your PR also meets following requirements:
- [ ] Pull Request requires a change to the documentation
- [ ] Documentation have been updated accordingly
- [x] Tests were added to cover changes

For new features ensure as well:
- [x] Analysis was done
- [x] Test Plan has been done
- [ ] Tests were verified in advance

### CODE REVIEW

**Comment by TomasHofman** [comment]
Eh, Wildfly 15 released already, so I need to rework this and create new management version.

**Comment by TomasHofman** [comment]
Updated.

**Comment by ehsavoie** [comment]
Could you please create a separate PR (and Jira) for the schema version bump ?

**Comment by TomasHofman** [comment]
Rebased.

**Comment by bstansberry** [comment]
@ehsavoie Please review using github review UI.

File: messaging-activemq/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties [hunk 1]
```diff
@@ -381,6 +381,7 @@ server.journal-min-files=How many journal files to pre-create.
 server.journal-node-manager-store-table=Name of the JDBC table to store the node manager.
 server.journal-page-store-table=Name of the JDBC table to store pages.
 server.journal-pool-files=The number of journal files that can be reused. ActiveMQ will create as many files as needed however when reclaiming files it will shrink back to the value (-1 means no limit).
+server.journal-file-open-timeout=The timeout (in seconds) for opening journal files.
```
Comment (bstansberry) [inline messaging-activemq/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties]: What happens if the timeout is exceeded?

Comment (TomasHofman) [reply messaging-activemq/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties]: Value <= 0 means fail immediately. I updated the description.

**Comment by TomasHofman** [comment]
Retest this please.

File: messaging-activemq/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties (hunk 1 above)
Comment (TomasHofman) [reply messaging-activemq/src/main/resources/org/wildfly/extension/messaging/activemq/LocalDescriptions.properties]: Sorry, looks like I didn't read the question carefully. If journal file cannot be opened, the ActiveMQ server shuts down.

**Comment by TomasHofman** [comment]
Retest this please.

**Review by ehsavoie — APPROVED** [review]

**Comment by ehsavoie** [comment]
retest this please

**Review by jamezp — APPROVED** [review]

**Comment by bstansberry** [comment]
retest this please

**Pull request merged**